Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift

```diff
@@ -578,3 +613,28 @@ private class DisplayURLLabel: UILabel {
     super.drawText(in: rect)
   }
 }
+
+extension UIImage {
+  /// Return a square gradient image
+  static func braveGradient(with frame: CGRect, darken: CGFloat? = nil) -> UIImage? {
+    let colors = [UIColor(rgb: 0xFA7250), UIColor(rgb: 0xFF1893), UIColor(rgb: 0xA78AFF)]
+    let layer = CAGradientLayer()
+    layer.backgroundColor = UIColor.black.cgColor
+    layer.frame = frame
+    layer.colors = colors.map({ color in
+      if let darken = darken {
+        return color.withAlphaComponent(darken).cgColor
+      } else {
+        return color.cgColor
+      }
+    })
+    layer.locations = [0, 0.5, 1]
+    layer.startPoint = CGPoint(x: 1, y: 1)
+    layer.endPoint = CGPoint(x: 0, y: 0)
+    UIGraphicsBeginImageContext(CGSize(width: frame.width, height: frame.height))
+    layer.render(in: UIGraphicsGetCurrentContext()!)
+    let image = UIGraphicsGetImageFromCurrentImageContext()
+    UIGraphicsEndImageContext()
+    return image
+  }
+}
```

Comment:
This doesn't seem like the right solution. Consider subclassing a UIControl/UIButton instead